stumpy/stamp.py

```diff
@@ -59,20 +61,22 @@ def mass(Q, T, M_T, Σ_T, trivial_idx=None, excl_zone=0, left=False, right=False
         if D[:trivial_idx].size:
             IL = np.argmin(D[:trivial_idx])
             PL = D[IL]
-        if zone_start <= IL < zone_stop:
-            IL = -1
+            if PL == np.inf or zone_start <= IL < zone_stop:
+                IL = -1
 
         IR = -1
         PR = np.inf
-        if D[trivial_idx:].size:
-            IR = trivial_idx + np.argmin(D[trivial_idx:])
+        if D[trivial_idx + 1 :].size:
```

Comment:
What is this "+1" for? Same question below

Reply:
Because the right profile index is the lowest index to the right of the trivial index, and D[trivial_idx] is always the lowest value (=0) in D[trivial_idx:]

Reply:
IMHO, I think the original logic is cleaner. Can you think of a scenario where, given the same input, the output would be different between the two versions of the code? If not, then I recommend we stick with the original version as the `+1` requires some mental gymnastics and isn't obvious (at least for "future Sean").

Also, if we change it here, I think there are other places within the code that we'll need to update as well.